packages/transformers/elm/src/ElmTransformer.js

```diff
@@ -0,0 +1,125 @@
+import {Transformer} from '@parcel/plugin';
+import commandExists from 'command-exists';
+import path from 'path';
+import {minify} from 'terser';
+import ThrowableDiagnostic from '@parcel/diagnostic';
+
+export default new Transformer({
+  async transform({asset, options}) {
+    const elm = await setupElmCompiler(asset, options);
+
+    const config = {
+      cwd: path.dirname(asset.filePath),
+      debug: !options.env.PARCEL_ELM_NO_DEBUG && options.mode !== 'production',
+      optimize: asset.env.minify,
+    };
+    asset.invalidateOnEnvChange('PARCEL_ELM_NO_DEBUG');
+    ResolveDependenciesFromAsset(elm, asset);
+
+    let code = await compileToString(elm, asset, config);
+    if (options.hot) {
+      code = await injectHotModuleReloadRuntime(code, asset.filePath, options);
+    }
+    if (config.optimize) code = await minifyElmOutput(code);
+
+    asset.type = 'js';
+    asset.setCode(code);
+    return [asset];
+  },
+});
+
+async function setupElmCompiler(asset, options) {
+  ensureElmIsInstalled();
+  await ensureElmJson(asset);
+  return options.packageManager.require('node-elm-compiler', asset.filePath, {
+    autoinstall: options.autoinstall,
+    saveDev: true,
+  });
+}
+
+function ensureElmIsInstalled() {
+  if (!commandExists.sync('elm')) {
+    throw new ThrowableDiagnostic({
+      diagnostic: {
+        message: "Can't find 'elm' binary.",
+        hints: [
+          "You can add it as an dependency for your project by running 'npm add -D elm' or 'npm add -D elm'.",
```

Comment:
One of there should be Yarn instead of npm twice 😄

Reply:
True :sweat_smile: